src/main/scala/piconot/external/parser/Parser.scala

```diff
@@ -0,0 +1,66 @@
+package  piconot.external.PicoDSL.parser
+
+import scala.util.parsing.combinator._
+import piconot.external.PicoDSL.ir._
+import picolib.semantics._
+
+
+object PicoParser extends JavaTokenParsers with PackratParsers {
+
+    // parsing interface
+    def apply(s: String): ParseResult[ProgramAST] = parseAll(program, removeComments(s))
+    
+    // removes # and everything after it on each line
+    def removeComments(s: String) = s.split("\n").map((s: String) => s.split("#")(0)).mkString("\n")
+    
```

Comment:
Love that you made comments work in the external DSL, that was one of the things we didn't get to.